File: advent_of_code/typescript/src/2022/day01.ts
import { getInputChallenge } from "../services/adventofcode";

getInputChallenge({ year: 2022, day: 1 })
  .then((res) => {
    // solvePuzzle(res.data)
    const data : string[] = res.data.split("\n");
    const result: number[] = [];
    let sum = 0;
    data.forEach((e: string) => {
      if (e.trim() === "") {
        result.push(sum);
        sum = 0;
        return;
      }
      sum += Number(e);
    });

    result.sort((a, b) => Number(b) - Number(a));
    const first = result[0];
    const total = first + result[1] + result[2];
    console.log({ first, total });
  })
  .catch((err) => console.log({ err }));


  function solvePuzzle(input: string) {
    const elves = input.split('\n\n'); // Split input into an array of strings, one string per Elf
    let maxCalories = 0;
    let maxElf = "";

    for (const elf of elves) {
        const items = elf.split('\n').filter(i => i); // Split each Elf's inventory into an array of items and remove empty strings
        const calories = items.map(i => parseInt(i, 10)).reduce((a, b) => a + b, 0); // Parse each item as a number and sum them up
        if (calories > maxCalories) {
            maxCalories = calories;
            maxElf = elf;
        }
    }

    console.log(`The Elf carrying the most Calories is: ${maxElf}`);
    console.log(`Total Calories: ${maxCalories}`);
}

File: advent_of_code/rust/aoc/src/years/mod_2022/day_01.rs
use aoc::get_input_example;

pub fn run() {
  let data = get_input_example(2022, 1);
  let data = data.split("\n");
  let mut sums : Vec<i32> = Vec::new();
  let mut sum  = 0;
  for line in data.clone() {
    if line.is_empty() {
      sums.push(sum);
      sum = 0;
      continue
    }
    sum += line.parse::<i32>().unwrap();
  }
  sums.sort();
  sums.reverse();
  println!("2022, day 01, first {:?}, total {:?}", sums[0], sums[0] + sums[1] + sums[2])
}